spec/Classes.tex

```diff
@@ -784,12 +787,54 @@ \subsection{The Method Receiver and the {\em this} Argument}
 is \chpl{c1}.
 \end{chapelexample}
 
-The optional \sntx{this-intent} declares the intent by which the receiver
-argument will be passed.  When the value of \sntx{this-intent} is \chpl{param},
-it implies that the function can only be applied to param objects of the given
-type binding.  If the value is \chpl{ref}, the argument will be passed by
-reference---allowing modification of \chpl{this}.  Otherwise, the argument will
-be passed with the blank intent as specified in \rsec{The_Blank_Intent}.
+The optional \sntx{this-intent} declares how the receiver argument is
+constrained or will be passed for a given method call.
+
```

Comment:
Another thought on all of this in the context of the code review.  I recently code reviewed some work Lydia
did for public/private modules.  Her commit included a fantastic collection of tests that clarified the behavior
of the current implementation.

For this work I wouldn't have picked up on this issue if I hadn't asked about an update to the Spec.  I just
scanned your tests again and I don't think there's a test that clarifies whether or not one should be able
to invoke a type method on a value with that type.